packages/astro/src/types/public/integrations.ts

```diff
@@ -225,13 +217,18 @@ export interface BaseIntegrationHooks {
 	'astro:build:done': (options: {
 		pages: { pathname: string }[];
 		dir: URL;
+		/** @deprecated Use `routes` from `astro:routes:resolved` instead */
 		routes: IntegrationRouteData[];
 		logger: AstroIntegrationLogger;
 	}) => void | Promise<void>;
 	'astro:route:setup': (options: {
 		route: RouteOptions;
 		logger: AstroIntegrationLogger;
 	}) => void | Promise<void>;
+	'astro:routes:resolved': (options: {
+		routes: Array<IntegrationResolvedRoute>;
```

Comment:
For consistency with the other arrays in the type
```suggestion
		routes: IntegrationResolvedRoute[];
```